### PR TITLE
Expose the SerialExecutionContext

### PR DESCRIPTION
**Pull request by bryce-anderson**
It's a useful type, so we're going to make it public.

### CODE REVIEW

File: core/src/main/scala/org/http4s/blaze/util/Actor.scala [hunk 1]
```diff
@@ -47,7 +51,7 @@ private[blaze] abstract class Actor[M](ec: ExecutionContext) {
     override def run(): Unit = {
       @tailrec
       def go(i: Int, next: Node): Unit =
-        if (i >= MaxIterations) reSchedule(next)
+        if (i >= DefaultMaxIterations) reSchedule(next)
```
[the remark this answers is not in the document]
Comment (bryce-anderson) [reply core/src/main/scala/org/http4s/blaze/util/Actor.scala]: Yes. 🤦‍♂️

File: core/src/main/scala/org/http4s/blaze/util/SerialExecutionContext.scala [hunk 2]
```diff
@@ -4,10 +4,19 @@ import scala.concurrent.ExecutionContext
...
+  * happens-before relationship with subsequent tasks, meaning mutations performed
+  * in a task are observed by all sequent tasks.
+  *
+  * @param parent `ExecutionContext` with which to perform the work, which may consist
+  *              of many tasks queued in the `SerialExecutionContext`.
+  * @param maxIterationsBeforeReschedule Maximum number of tasks to execute before
+  *                                      rescheduling work in the `parent`.
   */
-private[blaze] class SerialExecutionContext(parent: ExecutionContext) extends ExecutionContext {
+class SerialExecutionContext(
```
[the remark this answers is not in the document]
Comment (bryce-anderson) [reply core/src/main/scala/org/http4s/blaze/util/SerialExecutionContext.scala]: `trampoline` and `direct` are _instances_ in the `Execution` object. I think the closer analogy is the `TickWheelExecutor`, which is also in org.http4s.blaze.util`.

[the remark this answers is not in the document]
Comment (jmcardon) [reply core/src/main/scala/org/http4s/blaze/util/SerialExecutionContext.scala]: :+1:. That's a good point.

[the remark this answers is not in the document]
Comment (rossabaker) [reply core/src/main/scala/org/http4s/blaze/util/SerialExecutionContext.scala]: Is this ever used as a type, or could it be `def serial(ExecutionContext, Int): ExecutionContext`?

**Comment by bryce-anderson** [comment]
I'm planning on merging this sometime this weekend, barring objections.

**Comment by rossabaker** [comment]
Are we waiting on this one for any reason?

**Comment by jmcardon** [comment]
I was kinda just waiting on bryce to merge his own pr

**Pull request merged**